src/Gui/DlgPreferencesImp.cpp

```diff
@@ -471,6 +477,34 @@ void DlgPreferencesImp::applyChanges()
     }
 }
 
+void DlgPreferencesImp::isRebootRequired()
+{
+    if (rebootRequired) {
+        QMessageBox* restartBox = new QMessageBox();
+        restartBox->setIcon(QMessageBox::Warning);
+        restartBox->setWindowTitle(tr("Restart required"));
+        restartBox->setText(tr("You must restart FreeCAD for changes to take effect."));
+        restartBox->setStandardButtons(QMessageBox::Ok | QMessageBox::Cancel);
+        restartBox->setDefaultButton(QMessageBox::Cancel);
+        auto okBtn = restartBox->button(QMessageBox::Ok);
+        auto cancelBtn = restartBox->button(QMessageBox::Cancel);
+        okBtn->setText(tr("Restart now"));
+        cancelBtn->setText(tr("Restart later"));
+
+        int exec = restartBox->exec();
+
+        if (exec == QMessageBox::Ok) {
+            //restart FreeCAD after a delay to give time to this dialog to close
+            QTimer::singleShot(1000, []() 
+            {
+                QStringList args = QApplication::arguments();
+                if (getMainWindow()->close())
+                    QProcess::startDetached(QApplication::applicationFilePath(), args);

```

Comment:
Passing the unchanged `args` is incorrect because the first element is always the path to the executable. So, the first element of the list must be removed.

Reply:
I copied that from the addon manager. So there is probably the same problem there. Should it be fixed in the same PR?

Reply:
Note adding `args.pop_front();` removes correctly the `File format not supported: /home/user/Projects/build_clang/bin/FreeCAD`  message on restart.